src/app.py

```diff
@@ -1263,6 +1269,545 @@ def allowable_edit_states(hmuuid):
+        file_does_not_exist_msg = "No selected file"
+    if file_does_not_exist is True:
+        response_body = {"status": "fail", "message": file_does_not_exist_msg}
+        return Response(json.dumps(response_body, sort_keys=True), 400, mimetype='application/json')
+    if file_does_not_exist is False:
+        unsuccessful_upload = False
+        unsuccessful_upload_msg = ''
+        try:
+            temp_id = file_upload_helper_instance.save_temp_file(file)
+        except Exception as e:
+            unsuccessful_upload = True
+            unsuccessful_upload_msg = str(e)
+        # uses csv.DictReader to add functionality to tsv file. Can do operations on rows and headers.
+        records = []
+        headers = []
+        file_location = commons_file_helper.ensureTrailingSlash(app.config['FILE_UPLOAD_TEMP_DIR']) + temp_id + os.sep + file.filename
+        with open(file_location, newline='') as tsvfile:
+            reader = csv.DictReader(tsvfile, delimiter='\t')
+            first = True
+            for row in reader:
+                data_row = {}
+                for key in row.keys():
+                    if first: headers.append(key)
+                    data_row[key] = row[key]
+                records.append(data_row)
+                if first: first = False
+        validfile = validate_donors(headers, records)
+        if validfile == True:
+            if unsuccessful_upload is False:
+                valid_response = {}
+                #valid_response['Response'] = "Tsv file valid. File uploaded succesffuly"
+                valid_response['temp_id'] = temp_id
+                return Response(json.dumps(valid_response, sort_keys=True), 201, mimetype='application/json')
+                #return jsonify(valid_response)
+            if unsuccessful_upload is True:
+                msg = "Failed to upload files"
+                logger.exception(msg)
+                internal_server_error(msg)
+                response_body = {"status": "error", "data": {"File valid but upload failed": unsuccessful_upload_msg}}
+                return Response(json.dumps(response_body, sort_keys=True), 500, mimetype='application/json')
+        if type(validfile) == list:
+            return_validfile = {}
+            error_num = 0
+            for item in validfile:
+                return_validfile[str(error_num)] = str(item)
+                error_num = error_num + 1
+            #return_validfile = ', '.join(validfile)
+            response_body = {"status": "fail", "data": return_validfile}
+            return Response(json.dumps(response_body, sort_keys=True), 400, mimetype='application/json')  # The exact format of the return to be determined
+
+
+@app.route('/donors/bulk', methods=['POST'])
+def create_donors_from_bulk():
+    request_data = request.get_json()
+    token = str(request.headers["AUTHORIZATION"])[7:]
+    header = {'Authorization': 'Bearer ' + token}
+    temp_id = request_data['temp_id']
+    include_group = False
```

Comment:
No need to use this `include_group` flag, simply check the `group_uuid` to determine if to include. Something like below:

```
group_uuid = None

...

if group_uuid is not None:
    item['group_uuid'] = group_uuid
```